client/lib/happychat/connection.js

```diff
@@ -20,7 +20,10 @@ class Connection extends EventEmitter {
 				const socket = new IO( url );
 				socket
 					.once( 'connect', () => debug( 'connected' ) )
-					.on( 'init', () => resolve( socket ) )
+					.on( 'init', () => {
```

Comment:
I don't think either the base Socket.io or the Happychat service emit a `connected` event. `connect` is a core Socket.io event indicating a connection has been made, but `happychat-service` isn't actually ready until [it emits an `init` event](https://github.com/Automattic/happychat-service/blob/master/src/state/middlewares/socket-io/agents.js#L41) (which happens after it receives the `connect` event).

Ideally, I'd imagine we'd want `happychat-service` to suppress a `connect` event until it's actually ready for user interaction, rather than emit a `connect` and then later an `init` (which IMO is a bit confusing naming). I haven't dug in deep but it looks like HC service needs to authorize the user after `connect`.

You can see from the previous implementation that the `init` listener is what resolves the `connection.open()` promise, so in the previous implementation this is the event that triggered `dispatch( setChatConnected() );`. That's why I'm using it to emit a `connected` event here (so that in the middleware interface the listener names make a bit more sense). The dispatch should occur at the same place in the connection lifecycle it did before.